### ISSUE TITLE
Questions about Active Directory

### ISSUE THREAD

**Issue by tomkuipers1402** [obiba/agate#552]
Hi!
I'm connecting AD to Agate, but I have a few questions:
- Is it possible to use a ldaps instead of ldap, e.g., ldaps://localhost:636?
- I have setup AD in Agate now, but it's not working yet (when I try to login with my LUMC account it searches for the account in Agate users, but not in AD). I'm not getting much info from the logs. Is there a way I can verify that Agate is making an ok connection?

I got the AD working in KeyCloak as a test, but not able to get it to work in Agate. The logs only provide this when saving AD realm:
2024-07-22 09:49:57,745 INFO  org.obiba.agate.security.SecurityManagerFactory - Adding realm 'ldapTEST' to session manager.
But no confirmation of actually making a successful connection.
In KeyCloak I've set:
- Connection URL
- Bind DN (e.g., CN=USERNAME,OU=.........,DC=........)
- Bind credentials (Password of USERNAME)
- Users DN: OU=.........,DC=.........

How would this translate to Agate? Should "System User Name" just contain username or also CN=USERNAME,etc.?
Search filter is required, I use: ((objectClass=*)) for now
Search base is set the same as "Users DN" in KeyCloak
Hopefully you can guide me in the right direction here!

Thanks!

**Comment by tomkuipers1402** [comment]
Hi, hopefully you have time to look into this.
Best, Tom

**Comment by ymarcon** [comment]
Yes @kazoompa has started looking into this. It is a bit tedious to setup an environment to reproduce this issue (need a VM with AD), but the task is in progress.

**Comment by tomkuipers1402** [comment]
Ok great, thanks for the update!

**Comment by kazoompa** [comment]
Hi,

I think you need to look into the AD logs to see what  Agate sends for searching a user and see if it's not an issue of not finding it in AD or if it returns a user and itdoesn't match with any of Agate usernames ou emails. Agate usernames are usually with no spaces and CN may contain names that have spaces (e.g. Full Name). 

Ldaps should work as long as your server is properly setup and the port is available.

**Comment by tomkuipers1402** [comment]
Thanks for taking the time!
I'm still not able to login. I can reach the ldap server (ldaps not yet), so I continued with ldap for now, but good to know that ldaps is available. With ldapsearch I'm able to read the AD on the server. Once set & active in Agate, I can't login. When I try to login directly, it just says: 
```
2024-09-02 09:38:01,906 INFO  org.obiba.agate.web.rest.security.SessionsResource - Authentication failure of user 'MYUSERNAME@lumc.nl' at ip: '10.161.107.178': No account information found for authentication token [org.apache.shiro.authc.UsernamePasswordToken - MYUSERNAME@lumc.nl, rememberMe=false] by this Authenticator instance.  Please check that it is configured correctly.
```
I also tried to register before login (I set the AD as Sign up realm) with my LUMC email, but then it still is listed as "default realm".
What should I consider my username for Agate/Opal for AD users? The values from `userPrincipalName`?
Should an AD user first register in Agate? Or can I turn off Sign Up realm? What would be the use of Sign Up realm in Agate?
Once I know this, I can reach out to our IT & check the AD logs.
